zebra/zebra_mlag.c

```diff
@@ -78,8 +616,29 @@ DEFPY_HIDDEN (test_mlag,
 			   mlag_role2str(orig, buf1, sizeof(buf1)),
 			   mlag_role2str(orig, buf2, sizeof(buf2)));
 
-	if (orig != zrouter.mlag_info.role)
+	if (orig != zrouter.mlag_info.role) {
 		zsend_capabilities_all_clients();
+		if (zrouter.mlag_info.role != MLAG_ROLE_NONE) {
+			if (zrouter.mlag_info.clients_interested_cnt == 0
+			    && test_mlag_in_progress == false) {
+				frr_with_mutex (
+					&zrouter.mlag_info.mlag_th_mtx) {
```

Comment:
so ... this appears to be recursively locking this mutex that I was asking about:
lock mlag_th_mtx
call `spawn_pthread`
calls `zebra_mlag_signal_write_thread()`
takes the mutex again

Reply:
actually this will be accessed only in main thread, the only variable that was shared is "zrouter.mlag_info.th_master", and MLAG thread is copying this pointer to Local variable, it will never access this Global pointer. done some cleanup here. to make sure MLAG thread will not access any Globals in "zebra_mlag_signal_write_thread"